src/Keboola/StorageApi/Client.php

```diff
@@ -1330,10 +1331,29 @@ public function uploadFile($filePath, FileUploadOptions $options)
                 $result['name']
             );
 
+            $cnt = 0;
             do {
                 try {
+                    $cnt++;
                     $s3result = $uploader->upload();
                 } catch (MultipartUploadException $e) {
+                    $this->log('Multipart upload failed: ' . $e->getMessage());
+                    if ($cnt > 10) {
+                        $params = $e->getState()->getId();
+                        if (!empty($params['UploadId'])) {
+                            try {
+                                $s3Client->abortMultipartUpload($params);
```

Comment:
na abortování bych se vybodl, tím si jen přiděláme starosti. 

> As a result, it might be necessary to abort a given multipart upload multiple times in order to completely free all storage consumed by all parts.

https://docs.aws.amazon.com/AmazonS3/latest/API/mpUploadAbort.html

Reply:
yes, and also there is `To verify that all parts have been removed, so you don't get charged for the part storage, you should call the List Parts operation and ensure the parts list is empty.`
which sounds like a good idea too.